### PR TITLE
fix: cleanup volume-snapshot backup failure

### LINKED ISSUE #4778: delete a invalid backup get stuck when there is no backupmeta there
## Bug Report

**What version of Kubernetes are you using?**
`[root@kvm-dev ~]# kubectl version
Client Version: version.Info{Major:"1", Minor:"22+", GitVersion:"v1.22.6-eks-7d68063", GitCommit:"f24e667e49fb137336f7b064dba897beed639bad", GitTreeState:"clean", BuildDate:"2022-02-23T19:32:14Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22+", GitVersion:"v1.22.15-eks-fb459a0", GitCommit:"be82fa628e60d024275efaa239bfe53a9119c2d9", GitTreeState:"clean", BuildDate:"2022-10-24T20:33:23Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}`

**What version of TiDB Operator are you using?**
version: 1.4.0

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
gp3

**What's the status of the TiDB cluster pods?**
image

**What did you do?**

**What did you expect to see?**
delete job won't work, it get stuck
**What did you see instead?**

### PR DESCRIPTION
**Pull request by fengou1**
### What problem does this PR solve?
an incomplete backup can not be deleted

close #4778
### What is changed and how does it work?
an incomplete backup without the backup meta-file block the cleanup job

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
1. launch a backup
2. wait backup job is running
3. cancel the backup job, the delete job will be triggered and get stuck

- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: cmd/backup-manager/app/clean/clean.go [hunk 1]
```diff
@@ -81,6 +81,12 @@ func (bo *Options) deleteSnapshotsAndBackupMeta(ctx context.Context, backup *v1a
 		_ = os.Remove(metaFile)
 	}()
 
+	if _, err := os.Stat(metaFile); err != nil {
```
Comment (gozssky) [inline cmd/backup-manager/app/clean/clean.go:84]: Why this error doesn't report on line 76?

**Review by WangLe1321 — APPROVED** [review]
LGTM

**Review by gozssky — APPROVED** [review]

**Pull request merged**